frappe/model/delete_doc.py

```diff
@@ -140,10 +139,18 @@ def get_table_fields(field_doctype):
 	for t in list(set(tables)):
 		if t not in ignore_doctypes:
 			frappe.db.sql("delete from `tab%s` where parenttype=%s and parent = %s" % (t, '%s', '%s'), (doctype, name))
+			
+def update_flags(doc, flags=None, ignore_permissions=False):
```

Comment:
But flags is also passed as argument in delete_doc function, which should be set in doc.flags